core/src/main/java/com/linecorp/armeria/server/healthcheck/HealthCheckServiceBuilder.java

```diff
@@ -250,13 +259,23 @@ public HealthCheckServiceBuilder updatable(HealthCheckUpdateHandler updateHandle
         return this;
     }
 
+    /**
+     * Sets whether the specified {@link ActionType} is enabled or not for the
+     * {@link #build() HealthCheckService}. All {@link ActionType}s are disabled by default.
+     */
+    public HealthCheckServiceBuilder transientServiceAction(ActionType actionType, boolean enable) {
+        transientServiceActions.put(requireNonNull(actionType, "actionType"), enable);
+        return this;
+    }
+
     /**
      * Returns a newly created {@link HealthCheckService} built from the properties specified so far.
      */
     public HealthCheckService build() {
         return new HealthCheckService(healthCheckers.build(),
                                       healthyResponse, unhealthyResponse,
                                       maxLongPollingTimeoutMillis, longPollingTimeoutJitterRate,
-                                      pingIntervalMillis, updateHandler);
+                                      pingIntervalMillis, updateHandler,
+                                      Maps.newEnumMap(ImmutableMap.copyOf(transientServiceActions)));
```

Comment:
Could use `Maps.immutableEnumMap(...)`?
```suggestion
                                      Maps.immutableEnumMap(transientServiceActions));
```

Reply:
This is only used for the builder. It doesn't create many objects. You can ignore this review. :-)